Druid-Tests/DRProductionBytecodeTest.class.st

```diff
@@ -88,31 +88,55 @@ DRProductionBytecodeTest >> testBytecodeDuplicateTop [
 { #category : #tests }
 DRProductionBytecodeTest >> testBytecodePushLiteralConstant [
 
-	| literalIndex method |
-	literalIndex := 0 "0 based...".
-	self compileBytecode: 32 + literalIndex selector: #pushLiteralConstantBytecode thenDo: [ :generator |
-		"Execute the druid's compiled code"
-		generator value.
-	
-		"Then return without druid's compiled code"
-		cogit ssTop popToReg: ReceiverResultReg.
-		cogit genUpArrowReturn ].
-
+	| literalIndex method compiledMethod |
 	method := methodBuilder
-		newMethod;
-		literals: { (memory integerObjectOf: 42) };
-		buildMethod.
+		          newMethod
+		          literals: { (memory integerObjectOf: 42) };
+		          buildMethod.
+	cogit methodObj: method.
 
-	self executePrimitiveWithReceiver: (memory integerObjectOf: 17)
-		withArguments: #()
-		method: method.
-	self assert: machineSimulator receiverRegisterValue equals: (memory integerObjectOf: 42)
+	self flag: #TODO. "Make it work!"
+	"compiledMethod := cogit cog: method selector: memory nilObject."
```

Comment:
The test now is working because  `cogit methodObj: method` is hardcoded.

That line is done in `cogit cog: method selector: memory nilObject` that is what the test should do ("really" compile the method, not just set the `methodObj`).

I tried, and I'm close to success, but not yet :(

Reply:
Fixed in next PR